### PR TITLE
added sources

### PR DESCRIPTION
**Pull request by panser81**

### CODE REVIEW

**Review by aperavoznikau-idt — CHANGES_REQUESTED** [review]

File: src/main/java/com/gmail/spanteleyko/web/controllers/RolesServlet.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+    private static final String VIEW_TEMPLATE_PATH = "/WEB-INF/jsp/roles.jsp";
+
+    @Override
+    protected void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+        request.getRequestDispatcher(VIEW_TEMPLATE_PATH).forward(request, response);
+    }
+
+    @Override
+    protected void doPost(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+
```
[the remark this answers is not in the document]
Comment (panser81) [reply src/main/java/com/gmail/spanteleyko/web/controllers/RolesServlet.java:19]: removed

File: src/main/java/com/gmail/spanteleyko/web/converters/RoleConverter.java [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+    public static List<RoleDTO> convertToDTO(List<Role> roles) {
+        if (roles == null) {
+            return new ArrayList<>();
+        }
+
+        List<RoleDTO> roleDTO = new ArrayList<>();
+
+        roles.forEach(role -> {
+            roleDTO.add(convert(role));
+        });
```
[the remark this answers is not in the document]
Comment (panser81) [reply src/main/java/com/gmail/spanteleyko/web/converters/RoleConverter.java:19]: used map

File: src/main/java/com/gmail/spanteleyko/web/converters/RoleConverter.java [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+import com.gmail.spanteleyko.web.models.RoleDTO;
+import com.gmail.spanteleyko.web.repositories.models.Role;
+
+import java.util.ArrayList;
+import java.util.List;
+
+public class RoleConverter {
+    public static List<RoleDTO> convertToDTO(List<Role> roles) {
+        if (roles == null) {
+            return new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (panser81) [reply src/main/java/com/gmail/spanteleyko/web/converters/RoleConverter.java:12]: done

File: src/main/java/com/gmail/spanteleyko/web/models/UserDTO.java [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+
+import java.sql.Date;
+import java.util.List;
+
+public class UserDTO {
+    private Integer id;
+    private String username;
+    private String password;
+    private Date createdBy;
+    private List<RoleDTO> roles;
```
[the remark this answers is not in the document]
Comment (panser81) [reply src/main/java/com/gmail/spanteleyko/web/models/UserDTO.java:11]: done

File: src/main/java/com/gmail/spanteleyko/web/repositories/models/User.java [hunk 5]
```diff
@@ -0,0 +1,52 @@
+package com.gmail.spanteleyko.web.repositories.models;
+
+import java.sql.Date;
+import java.util.List;
+
+public class User {
+    private Integer id;
```
[the remark this answers is not in the document]
Comment (panser81) [reply src/main/java/com/gmail/spanteleyko/web/repositories/models/User.java:7]: use Long

File: src/main/java/com/gmail/spanteleyko/web/services/impl/RoleServiceImpl.java [hunk 6]
```diff
@@ -0,0 +1,85 @@
...
+
+    @Override
+    public List<RoleDTO> get(int id) throws SQLException {
+        try (Connection connection = connectionRepository.getConnection()) {
+            connection.setAutoCommit(false);
+            List<Role> roles = roleRepository.get(connection, id);
+            return RoleConverter.convertToDTO(roles);
+        } catch (SQLException e) {
+            logger.error(e.getMessage(), e);
+        }
```
[the remark this answers is not in the document]
Comment (panser81) [reply src/main/java/com/gmail/spanteleyko/web/services/impl/RoleServiceImpl.java:47]: added transaction

File: src/main/java/com/gmail/spanteleyko/web/services/impl/RoleServiceImpl.java [hunk 7]
```diff
@@ -0,0 +1,85 @@
...
+    @Override
+    public List<RoleDTO> get() throws SQLException {
+        try (Connection connection = connectionRepository.getConnection()) {
+            connection.setAutoCommit(false);
+            List<Role> roles = roleRepository.get(connection);
+            return RoleConverter.convertToDTO(roles);
+        } catch (SQLException e) {
+            logger.error(e.getMessage(), e);
+        }
+        return new ArrayList<>();
```
[the remark this answers is not in the document]
Comment (panser81) [reply src/main/java/com/gmail/spanteleyko/web/services/impl/RoleServiceImpl.java:60]: modified

**Pull request merged**